hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -92,7 +93,8 @@ public void initialize() {
         super.initialize();
 
         mapNearCacheManager = mapServiceContext.getMapNearCacheManager();
-        nearCache = mapNearCacheManager.getOrCreateNearCache(name, mapConfig.getNearCacheConfig());
+        IMapDataStructureAdapter<K, V> adapter = new IMapDataStructureAdapter<K, V>(this);
+        nearCache = mapNearCacheManager.getOrCreateNearCache(name, mapConfig.getNearCacheConfig(), adapter);
```

Comment:
This is all the code it needs to implement the Near Cache preloader for `IMap` on a Hazelcast member :)